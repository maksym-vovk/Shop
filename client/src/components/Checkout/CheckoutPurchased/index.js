import React from 'react';
import {Link} from "react-router-dom";

import { CartButton } from "../../Cart/CartButton";

// styles
import './index.scss';

export const CheckoutPurchased = () => {
  return (
    <div className="container">
      <div className="purchase">
        <div className="purchase__block">
          <img className="purchase__logo" src="/static/img/Apple_Grey_Logo.png" alt="apple-logo"/>
          <h1 className="purchase__text">Thank you for purchase</h1>
          <Link to="/all">
            <CartButton state="Continue Shopping"/>
          </Link>
        </div>
      </div>
    </div>
  )
};