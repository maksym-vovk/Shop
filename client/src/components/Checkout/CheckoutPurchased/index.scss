.purchase {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;

  &__block {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom: 5%;
  }

  &__logo {
    width: 30%;
  }

  &__text {
    font-size: 30px;
  }
}

@media (min-width: 768px) and (max-width: 1024px) {
  .purchase {
    &__text {
      font-size: 26px;
    }

    &__block {
      padding-top: 35%;
    }
  }
}

@media (min-width: 481px) and (max-width: 767px) {
  .purchase {

    &__block {
      flex-direction: column;
      padding-top: 45%;
    }

    &__logo {
      width: 45%;
    }

    &__text {
      margin-bottom: 100px;
      font-size: 24px;
    }
  }
}

@media (min-width: 320px) and (max-width: 480px) {
  .purchase {

    &__block {
      flex-direction: column;
      padding-top: 65%;
    }

    &__logo {
      width: 57%;
    }

    &__text {
      margin-bottom: 100px;
      font-size: 20px;
      text-align: center;
    }
  }
}