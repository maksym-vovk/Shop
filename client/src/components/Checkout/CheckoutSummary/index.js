import React from 'react';
import {Link} from 'react-router-dom';

// styles
import './index.scss';

export const CheckoutSummary = ({removeSummary, totalPrice, totalItems, grandTotalPrice}) => {
  const checkBackground = (event) => {
    const target = event.target;
    if (target.classList.contains('background') || target.classList.contains('summary__cancel-image')) {
      removeSummary()
    }
  };
  return (
    <div onClick={checkBackground} className="background">
      <div className="summary">
        <div className="summary-container">
          <div className="summary__cancel">
            <img className="summary__cancel-image" src="../static/img/cancel.png" alt="cancel"/>
          </div>
          <div className="total">
            <h3 className="total__title">Your Oder Total</h3>
            <div className="total__items-section">
              <span className="total__quantity">Quantity: {totalItems} items</span>
              <Link className="total__edit" to="/cart">Edit bag</Link>
            </div>
            <div className="total__price">
              <div className="total__price-item">
                <span className="total__label">Subtotal</span>
                <span className="total__cost">{totalPrice}</span>
              </div>
              <div className="total__price-item">
                <span className="total__label">Shipping</span>
                <span className="total__cost">1000</span>
              </div>
            </div>
            <div className="total__sum-block">
              <span className="total__sum-label">Total</span>
              <span className="total__sum">{grandTotalPrice}</span>
            </div>
          </div>
          <div className="details">
            <h3 className="details__title">Your Order Details</h3>
            <p className="details__item">Shipment method</p>
            <p className="details__item">Delivery address</p>
            <p className="details__item">Payment</p>
          </div>
        </div>
      </div>
    </div>
  )
};