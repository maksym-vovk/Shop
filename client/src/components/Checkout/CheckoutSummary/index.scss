.background {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 10;
  display: flex;
  justify-content: flex-end;
  height: 100%;
  background-color: rgba(0, 0, 0, .5);
}

.summary {
  position: absolute;
  top: 25%;
  right: 34%;
  width: 30vw;
  height: 500px;
  background-color: #fff;
  border: 1px solid rgba(210, 215, 211, 1);

  &__cancel {
    position: absolute;
    right: 30px;
    width: 25px;
    height: 25px;
  }

  &__cancel-image {
    width: 100%;
    height: 100%;
  }
}

.summary-container {
  margin: 30px;
}

.total {
  display: flex;
  flex-direction: column;
  width: 100%;

  &__title {
    margin-top: 45px;
    padding: 20px 0;
    font-size: 20px;
    border-bottom: 1px solid darkgrey;
  }

  &__items-section {
    display: flex;
    justify-content: space-between;
    padding: 30px 0;
  }

  &__quantity {
    font-weight: bold;
    font-size: 17px;
  }

  &__edit {
    display: flex;
    align-self: flex-end;
    background-color: inherit;
  }

  &__price {
    display: flex;
    flex-direction: column;
    padding: 30px 0;
    font-size: 17px;
    border-top: 1px solid rgba(210, 215, 211, 1);
    border-bottom: 1px solid rgba(210, 215, 211, 1);
  }

  &__price-item {
    display: flex;
    justify-content: space-between;
  }

  &__label {
    margin: 5px 0;
  }

  &__sum-block {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 20px 0;
    font-weight: bold;
    font-size: 20px;
  }
}

.details {
  margin-top: 50px;

  &__title {
    padding: 20px 0;
    font-size: 20px;
    border-bottom: 1px solid darkgrey;
  }

  &__item {
    padding: 30px 0;
    color: darkgrey;
    border-bottom: 1px solid rgba(210, 215, 211, 1);
  }
}

@media (min-width: 1025px) and (max-width: 1280px) {
  .summary {
    width: 40vw;
    .total {
      &__title {
        margin-top: 30px;
      }
    }
    .details {
      margin-top: 0;

      &__item {
        padding: 20px 0;
      }
    }
  }
}

@media (min-width: 768px) and (max-width: 1024px) {
  .summary {
    width: 40vw;
    .total {
      &__title {
        margin-top: 30px;
      }
    }
    .details {
      margin-top: 0;

      &__item {
        padding: 20px 0;
      }
    }
  }
}




