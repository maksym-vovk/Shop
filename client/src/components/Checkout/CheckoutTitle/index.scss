.check-out {
  &__title-block {
    display: flex;
    justify-content: space-between;
    padding: 20px 0;
    border-bottom: 1px solid rgba(210, 215, 211, 1);
  }
  &__button {
    color: darkred;
    font-size: 16px;
    text-decoration: underline;
    background-color: inherit;
    border: none;
    cursor: pointer;
  }
}


/*
//  ##Device = Low Resolution Tablets, Mobiles (Landscape)
//  ##Screen = B/w 481px to 767px
//*/

@media (min-width: 481px) and (max-width: 767px) {
  .check-out {
    &__title-block {
      font-size: 13px;
    }
  }
}

/*
  ##Device = Most of the Smartphones Mobiles (Portrait)
  ##Screen = B/w 320px to 479px
*/
@media (min-width: 320px) and (max-width: 480px) {
  .check-out {
    &__title-block {
      font-size: 13px;
    }
  }
}