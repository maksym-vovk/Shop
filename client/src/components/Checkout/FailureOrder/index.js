import React from 'react';

import './index.scss';

export const FailureOrder = ({removeErrorMessage}) => {
  const checkBackground = (event) => {
    const target = event.target;
    console.log(target);
    if (target.classList.contains('background') || target.classList.contains('failure-modal__cancel-image')) {
      removeErrorMessage(false)
    }
  };

  return (
    <div onClick={checkBackground} className="background">
      <div className="failure-modal">
        <div className="failure-modal__cancel">
          <img className="failure-modal__cancel-image" src="../static/img/cancel.png" alt="cancel"/>
        </div>
        <div className="failure-modal__failure-image">
          <img className="failure-modal__img" src="../static/img/sad-emoticon.png" alt="sad"/>
        </div>
        <p className="failure-modal__text">Oops! Something went wrong...</p>
      </div>
    </div>
  )
};