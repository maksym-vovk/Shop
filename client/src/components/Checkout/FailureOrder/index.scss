.failure-modal {
  position: absolute;
  top: 25%;
  right: 36%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 30vw;
  height: 500px;
  background-color: #fff;
  border: 1px solid rgba(210, 215, 211, 1);

  &__cancel {
    position: absolute;
    top: 20px;
    right: 30px;
    width: 25px;
    height: 25px;
  }

  &__cancel-image {
    width: 100%;
    height: 100%;
  }

  &__text {
    margin-top: 30px;
    font-size: 25px;
  }

  &__failure-image {
    width: 60px;
    height: 60px;
  }

  &__img {
    width: 100%;
    height: 100%;
  }
}

@media (min-width: 1025px) and (max-width: 1280px) {
  .failure-modal {
    right: 25%;
    width: 500px;
  }
}

@media (min-width: 768px) and (max-width: 1024px) {
  .failure-modal {
    right: 17%;
    width: 500px;
  }
}

@media (min-width: 481px) and (max-width: 767px) {
  .failure-modal {
    right: 0;
    width: 100%;

    &__text {
      font-size: 20px;
    }
  }
}

@media (min-width: 320px) and (max-width: 480px) {
  .failure-modal {
    right: 0;
    width: 100%;

    &__text {
      font-size: 20px;
    }
  }
}