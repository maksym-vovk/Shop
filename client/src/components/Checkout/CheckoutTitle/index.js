import React, {useState} from 'react';
import {CheckoutSummary} from '../CheckoutSummary';

import './index.scss';

export const CheckoutTitle = ({totalPrice, totalItems, data, grandTotalPrice}) => {
  const [summary, setSummary] = useState(false);
  const changeSummaryStatus = () => {
    setSummary(!summary)
  };

  return (
    <div>
      {
        summary
          ? <CheckoutSummary
            removeSummary={changeSummaryStatus}
            totalPrice={totalPrice}
            data={data}
            totalItems={totalItems}
            grandTotalPrice={grandTotalPrice}
          />
          : null
      }
      <div className="check-out">
        <div className="check-out__title-block">
          <h1 className="check-out__title">Checkout</h1>
          <button onClick={() => setSummary(true)} className="check-out__button">Show Order Summary: ${totalPrice}</button>
        </div>
      </div>
    </div>
  )
};