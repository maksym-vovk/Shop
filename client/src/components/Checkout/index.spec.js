import React from 'react';
import { CheckoutPurchased } from './CheckoutPurchased';
import { CheckoutSummary } from './CheckoutSummary';
import { CheckoutTitle } from './CheckoutTitle';
import { FailureOrder } from './FailureOrder';

describe('Checkout components', () => {
  it('CheckoutPurchased component render', () => {
    const wrapper = shallow(<CheckoutPurchased />);
    expect(wrapper).toMatchSnapshot();
  });
  it('CheckoutSummary component render', () => {
    const wrapper = shallow(<CheckoutSummary />);
    expect(wrapper).toMatchSnapshot();
  });
  it('CheckoutTitle component render', () => {
    const wrapper = shallow(<CheckoutTitle />);
    expect(wrapper).toMatchSnapshot();
  });
  it('FailureOrder component render', () => {
    const wrapper = shallow(<FailureOrder />);
    expect(wrapper).toMatchSnapshot();
  });
});