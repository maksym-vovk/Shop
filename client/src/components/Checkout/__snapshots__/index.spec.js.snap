// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout components CheckoutPurchased component render 1`] = `ShallowWrapper {}`;

exports[`Checkout components CheckoutSummary component render 1`] = `ShallowWrapper {}`;

exports[`Checkout components CheckoutTitle component render 1`] = `ShallowWrapper {}`;

exports[`Checkout components FailureOrder component render 1`] = `ShallowWrapper {}`;
